### ISSUE TITLE
[Bug]: When disarmed via FOB, state doesn't update for group in home assistant

### ISSUE THREAD

**Issue by ArshSoni** [bvis/aegis-hass#287]
### Describe the bug

Group has been armed, and the area has been accessed and disarmed via the key fob (black/white), despite disarming, the state in HA doesn't update. 

FCM is configured and working with normal disarms via the ajax app or via HA

### Steps to reproduce

Arm group
Access the area
Disarm via fob
State not updating

### Expected behavior

Should update state in HA

### Home Assistant version

latest

### Integration version

Other / not listed

### Installation method

HACS

### App label (co-branded app)

ajax

### FCM push notifications configured?

Yes

### Relevant logs

```text

```

**Comment by bvis** [comment]
Thanks for the clear report. Working hypothesis: app/HA disarms arrive on one event family the integration already maps for per-group panels, while hub-originated actions (like a SpaceControl fob) likely arrive on a different family it doesn't map yet — which would match exactly the asymmetry you describe. To confirm before shipping a fix:

1. Which integration version are you running? (Settings → Devices & services → Aegis for Ajax shows it under the integration name.)
2. Enable debug logging for the integration (Settings → Devices & services → Aegis for Ajax → ⋮ → Enable debug logging), then arm the group and disarm it with the fob.
3. Paste the log lines from around the disarm — especially any line like `Push event parsed: event_type=… raw_tag=… group_id=…`. That `raw_tag` value tells us exactly which event the fob produces and whether the group context arrives with it.
4. Also useful: does the group panel correct itself later (within ~5 minutes, or after up to an hour), or does it stay wrong until you act on it again?

With that log this should be a quick fix.

**Comment by wip3out3r** [comment]
Adding a datapoint: same behavior here with a **scenario-driven group arm** (not a fob), so this does not look fob-specific. The common factor seems to be: group operations originating outside the app/HA do not trigger the immediate group state re-read.

**Setup**: v1.11.4-beta.3, group mode (2 groups, space "Home"), FCM configured and working (event entity updates instantly), SIA running in parallel.

**Timeline (2026-06-12, times UTC, from HA automation traces and recorder):**

```text
19:00:09.457  FCM security event received: event_type=disarm, device_type=SCENARIO, device_name=Inserisci GARAGE alle 21, device_id=00000002
19:00:09.5    alarm_control_panel.ajax_hub (space "Home"): disarmed -> armed_custom_bypass   (immediate, expected per #284 with mixed groups)
19:00:09 ...  alarm_control_panel.ajax_hub_garage (group): stays "disarmed"                  (stale)
19:09:11.068  alarm_control_panel.ajax_hub_garage (group): disarmed -> armed_away            (~9 minutes later, no new push at that time: looks like the internal periodic refresh picking it up)
```

So the push arrived and the space-level panel reacted instantly, but the immediate group re-read (#266 / #270 path) either did not run or returned stale group data for this scenario-driven operation. The group entity only caught up about 9 minutes later.

Two extra observations that may help narrow it down:

1. The FCM event for this scenario **arm** was delivered with `event_type=disarm` (with `device_type=SCENARIO` and the scenario name as `device_name`). The space transition at the same moment was disarmed -> armed_custom_bypass and the group ended up armed_away, so the operation was definitely an arm. If the security-refresh path filters on event type/direction, a mis-mapped event type could also explain why the group re-read did not fire.
2. Arms/disarms done from the app or from HA update the group panels within ~1s as expected, exactly as reported in the OP.

Happy to capture debug logs at the next occurrence if useful (the scenario runs nightly, so it is easy to reproduce here).

*Trace and recorder analysis assisted by Claude Code (Claude Fable 5).*

**Comment by bvis** [comment]
@wip3out3r excellent datapoint — it matches the pattern we're converging on: group operations originating outside the app/HA (fob, keypad, scenario) arrive on event families the integration doesn't yet map to the immediate group re-read, so the group panel waits for a periodic refresh. Your observation that the scenario arm was delivered as `event_type=disarm` is exactly the kind of mis-mapping we need to pin down. Since your scenario runs nightly:

1. Enable debug logging (Settings → Devices & services → Aegis for Ajax → ⋮ → Enable debug logging) before the scenario fires.
2. After it fires, disable debug logging (the log downloads automatically) and paste the lines around the event — especially any `Push event parsed: event_type=… raw_tag=… group_id=…` line. That `raw_tag` tells us which event the scenario produces and whether the group context arrives with it.

@ArshSoni the same debug capture for a fob disarm is still the missing piece on your side — with both tags this becomes one fix covering every origin.

**Comment by wip3out3r** [comment]
> @wip3out3r excellent datapoint — it matches the pattern we're converging on: group operations originating outside the app/HA (fob, keypad, scenario) arrive on event families the integration doesn't yet map to the immediate group re-read, so the group panel waits for a periodic refresh. Your observation that the scenario arm was delivered as event_type=disarm is exactly the kind of mis-mapping we need to pin down. Since your scenario runs nightly:
> 
> Enable debug logging (Settings → Devices & services → Aegis for Ajax → ⋮ → Enable debug logging) before the scenario fires.
> After it fires, disable debug logging (the log downloads automatically) and paste the lines around the event — especially any Push event parsed: event_type=… raw_tag=… group_id=… line. That raw_tag tells us which event the scenario produces and whether the group context arrives with it.
> @ArshSoni the same debug capture for a fob disarm is still the missing piece on your side — with both tags this becomes one fix covering every origin.

Could not resist testing **v1.11.4-beta.6** right away: I set up a one-off "arm GARAGE group" scenario on purpose and fired it at 01:00 tonight just for this test (no way I was waiting 20 hours for the nightly one). Result: **fixed**. The group panel now follows in **~165 ms** instead of waiting for the periodic refresh (it took ~9 minutes on beta.3).

**Debug capture around the scenario firing (log times are local, UTC+2; hub/notification ids redacted):**

```text
2026-06-13 01:00:02.970 DEBUG (MainThread) [custom_components.aegis_ajax.api.hts.client] RECV: type=0x08 seq=2881811 sender=00XXXXXX link=20 payload=50b
2026-06-13 01:00:02.970 DEBUG (MainThread) [custom_components.aegis_ajax.api.hts.client]   EVENT(0x08) payload hex: 0502052a0500000002052a05fdfd06360042ff9c6205fdfd01000205fdfd04000205fdfd07006a2c8f7105<text:3b>0a0a5f0c
2026-06-13 01:00:04.764 DEBUG (MainThread) [custom_components.aegis_ajax.notification] Push notification received: persistent_id=0:17813052XXXXXXXXX%XXXXXXXXXXXXXXXX
2026-06-13 01:00:05.113 DEBUG (MainThread) [custom_components.aegis_ajax.notification] Extracted notification_id: 5350435DXXXXXXXXXXXX
2026-06-13 01:00:05.114 DEBUG (MainThread) [custom_components.aegis_ajax.notification] Push event parsed: event_type=disarm raw_tag=space_disarmed group_id=None
2026-06-13 01:00:05.281 DEBUG (MainThread) [custom_components.aegis_ajax.coordinator] Finished fetching aegis_ajax data in 0.164 seconds (success: True)
```

HA state machine right after (times UTC, from recorder):

```text
23:00:05.116  event.ajax_hub_evento_di_sicurezza updated (FCM push, device_type=SCENARIO)
23:00:05.281  alarm_control_panel.ajax_hub (space): disarmed -> armed_custom_bypass
23:00:05.282  alarm_control_panel.ajax_hub_garage (group): disarmed -> armed_away
```

So, answering your question about which event the scenario produces: the scenario **group arm** is delivered with `raw_tag=space_disarmed` and `group_id=None` - no group context and even the wrong direction (it was an arm). The #295 snapshot nudge makes the panels converge regardless, which from the consumer side is what matters: with beta.6 the peripheral-originated 0x08 at 01:00:02.970 plus the FCM push at 01:00:05.114 both end with a fresh snapshot fetch and the group panel lands correct within ~165 ms.

One cosmetic leftover you may or may not care about: the event entity still exposes `event_type=disarm` for these scenario arms (same mis-mapped `space_disarmed` family), so anything keying on the event *direction* for non-app operations still sees the wrong verb. The panels are authoritative now, so this is minor for us.

Our notification pipeline behaved perfectly on beta.6: single notification, correct group, scenario name attributed.

Finally: issue reported yesterday evening, root cause hypothesized within the hour, fix shipped and confirmed in the field the next night. The dedication you put into this project is enormous and the whole Ajax + HA community is lucky to have it. Thank you!

*Captured and analyzed with Claude Code (Claude Fable 5).*

**Comment by bvis** [comment]
@wip3out3r this is the best kind of confirmation — thank you for the deliberate 01:00 test and the clean capture. ~165 ms vs ~9 minutes is exactly the outcome we were after, and your log answered the open question precisely: the scenario group arm comes through as `raw_tag=space_disarmed, group_id=None` — no group context and the wrong direction. That tells us mapping the hub `group_*` tag family would not have helped scenarios at all; what makes the panels converge is the authoritative snapshot re-read (#295), now also triggered by the peripheral-originated HTS event (#294). Since the panels are authoritative, the state is correct regardless of the bogus tag.

The cosmetic leftover you spotted is real and noted: the event entity still reports `event_type=disarm` for scenario arms because the `space_disarmed` tag itself is mis-mapped upstream. The panels don't rely on it anymore, so it's low-priority, but I'm tracking it as a separate follow-up to decode that tag's direction correctly. If you have automations keying on the event *verb* for non-app operations, hold off on those for scenario triggers until that lands.

And thank you for the kind words — captures and analysis like yours are what make a turnaround this fast possible. 🙏

**Comment by wip3out3r** [comment]
Thanks @bvis. I am covered on my side: nothing in my setup keys on the event verb for arm/disarm logic. My automations and dashboard read the per-zone `alarm_control_panel` panels, which beta.6 keeps authoritative, so the mis-mapped `event_type=disarm` on scenario arms has no functional impact here. I only flagged it for completeness, and I keep an internal rule to never rely on the event direction for non-app operations until the upstream decode lands.

If it helps when you get to decoding the `space_disarmed` direction: my nightly scenario reproduces it reliably, so I can capture a fresh debug log against any test build you want to point at the tag.

Appreciate the turnaround as always :)

**Comment by ArshSoni** [comment]
Hi, it seems with the latest beta update, this is working as expected now. It seems to be recognising fob updates correctly.

**Comment by bvis** [comment]
Thanks @ArshSoni for the report and for confirming the fix, and @wip3out3r for the detailed captures and for confirming the sibling scenario path. The per-group panel now follows fob/keypad/scenario operations via the authoritative snapshot re-read, shipped in v1.11.4-beta.6 (PRs #294 + #295).

Closing as resolved. One cosmetic leftover remains — scenario arms still surface `event_type=disarm` on the hub event entity (the per-zone panels are authoritative, so no functional impact); I'll track decoding that direction separately.
